### DISCUSSION TITLE
Documentation about OpenAPI errors content types is confusing

### DISCUSSION THREAD

**Discussion by cortopy** [salvo-rs/salvo#821] [Q&A] [answered]
In the [documentation about error handling with OpenAPI](https://salvo.rs/book/features/openapi.html#error-handling), it appears as if `StatusError` will be automatically generated as an `application/json` content type. However, it's by default a string since `StatusError` doesn't implement any Derive traits.

With the following code:

```rust
use salvo::http::{StatusCode, StatusError};
use salvo::oapi::{self, EndpointOutRegister, ToSchema};
use salvo::prelude::*;
use thiserror::Error;

#[derive(Error, Debug)]
pub enum ApiError {
    #[error("Unexpected error")]
    Unexpected,
}

#[async_trait]
impl Writer for ApiError {
    async fn write(mut self, _req: &mut Request, depot: &mut Depot, res: &mut Response) {
        res.render(StatusError::internal_server_error());
    }
}

impl EndpointOutRegister for ApiError {
    fn register(components: &mut oapi::Components, operation: &mut oapi::Operation) {
        operation.responses.insert(
            StatusCode::INTERNAL_SERVER_ERROR.as_str(),
            oapi::Response::new("Internal server error")
                .add_content("application/json", StatusError::to_schema(components)),
        );
    }
}
```

if the endpoint handler returns an error, the clients gets a `plain/text` response with the following body:

```
code: 500

name: Internal Server Error

brief: The server encountered an internal error while processing this request.

cause: There is no more detailed explanation.
```

This makes the OpenAPI specs inavlid since they say that 500 status code should come with json, but it doesn't. Is there anything I'm missing?

**Comment by chrislearn** [comment, marked as answer]
If you add http header accept="application/json" to the request, will it still return plain text?

**Comment by cortopy** [comment]
ah! that was it. Thanks @chrislearn. My endpoint returns `String` and Swagger UI was sending `Content-Type: plain/text`
